pkg/kubectl/cmd/attach.go

```diff
@@ -167,7 +167,7 @@ func (p *AttachOptions) Run() error {
 		if err != nil {
 			return err
 		}
-		if pod.Status.Phase != api.PodRunning {
+		if pod.Status.Phase != api.PodRunning && (pod.Status.Phase == api.PodPending && len(pod.Spec.InitContainers) < 1) {
```

Comment:
Yes, just those conditions.

On Fri, Jun 17, 2016 at 4:08 AM, Sylwester Brzęczkowski <
notifications@github.com> wrote:

> In pkg/kubectl/cmd/attach.go
> <https://github.com/kubernetes/kubernetes/pull/27541#discussion_r67473581>
> :
>
> > @@ -167,7 +167,7 @@ func (p *AttachOptions) Run() error {
> >  		if err != nil {
> >  			return err
> >  		}
> > -		if pod.Status.Phase != api.PodRunning {
> > +		if pod.Status.Phase != api.PodRunning && (pod.Status.Phase == api.PodPending && len(pod.Spec.InitContainers) < 1) {
>
> So should I change it to include conditions from
> https://github.com/kubernetes/kubernetes/pull/27359/files#diff-de6f7fc4d605130323893d0b7572045eR177
> ? @sttts <https://github.com/sttts> @smarterclayton
> <https://github.com/smarterclayton> ptal
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27541/files/3182d93c872757c8f639e5e18569c512bdbddb82#r67473581>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p0Vfb5MkXvDXFoI8wnlGkZ6zitXSks5qMlXvgaJpZM4I3cxi>
> .
>